### PR TITLE
fix ''event.preventDefault()' call does not change the 'event.defaultPrevented' property value ' (close #1588)

### LINKED ISSUE #1588: 'event.preventDefault()' call does not change the 'event.defaultPrevented' property value
Example to reproduce:
```js
var input = document.createElement('input');

    document.body.appendChild(input);

    input.addEventListener('keydown', function (e) {
        e.preventDefault();

        ok(e.defaultPrevented); // should be true
    });

    eventSimulator.keydown(input);
```

### PR DESCRIPTION
**Pull request by miherlosev**
@helen-dikareva @AndreyBelym 

Fix for https://github.com/DevExpress/testcafe/issues/2070

### CODE REVIEW

File: src/client/sandbox/event/simulator.js [hunk 1]
```diff
@@ -414,21 +403,18 @@ export default class EventSimulator {
...
-            // NOTE: the dispatchEvent method does not return false in the case when preventDefault
-            // was raised for events that were created with the KeyboardEvent constructor
-            if (browserWithNewEventsStyle) {
-                nativeMethods.objectDefineProperty.call(window.Object, ev, 'preventDefault', {
-                    get: () => () => {
-                        prevented = true;
+            // NOTE: the dispatchEvent method does not return false in the case when preventDefault method
+            // was called for events that were created with the KeyboardEvent constructor
+            if (this.browserWithNewEventsStyle) {
+                ev.preventDefault = function () {
```
[the remark this answers is not in the document]
Comment (miherlosev) [reply src/client/sandbox/event/simulator.js]: Unfortunately, we can not use this wrapper by architecture problem:
`EventSimulator` is a component of `EventSandbox`. And of `ElementEditingWatcher`, `StorageSandbox` and `UploadSandbox` too.
It turns out that we should add link to the created class (`EventSandbox`) from its parameter  (`EventSimulator`).
see https://github.com/DevExpress/testcafe-hammerhead/blob/master/src/client/sandbox/index.js#L57

So, it's better to duplicate code.

[the remark this answers is not in the document]
Comment (miherlosev) [reply src/client/sandbox/event/simulator.js]: I will add comment about it

[the remark this answers is not in the document]
Comment (AndreyBelym) [reply src/client/sandbox/event/simulator.js]: Got it, let's keep as it is now, but please, open an issue about refactoring this piece of code. It's a pity that we can't use more reliable prototype-level overload here.

[the remark this answers is not in the document]
Comment (miherlosev) [reply src/client/sandbox/event/simulator.js]: https://github.com/DevExpress/testcafe-hammerhead/issues/1600

**Pull request merged**